x-pack/plugins/observability/public/pages/alert_details/hooks/use_bulk_untrack_alerts.tsx

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { useMutation } from '@tanstack/react-query';
+import { INTERNAL_BASE_ALERTING_API_PATH } from '@kbn/alerting-plugin/common';
+import { useKibana } from '../../../utils/kibana_react';
+
+export const useBulkUntrackAlerts = () => {
```

Comment:
Is this logic from `triggers_actions_ui`? Is it possible to share the hook if that's the case? Maybe then we can get `http` and `toasts` as props to not reply on plugin's useKibana.

Reply:
I had first used the `useBulkUntrackAlerts` hook from `triggers_actions_ui`, but this hook is tightly coupled with Alerts Table, and depends on `AlertsTableQueryContext` [here](https://github.com/elastic/kibana/blob/cc7b89b50255f6511e0520e873990a6a2239dbfa/x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/hooks/use_bulk_untrack_alerts.tsx#L34). After discussing with @Zacqary, we decided to duplicate the hook in `observability` plugin where the hook will work without `AlertsTableQueryContext` context.

Since the functionality is same, I didn't change rest of the logic to reuse the same API and to be consistent with the hook from `triggers_actions_ui` in terms of success/failure messages.

Reply:
Just curious, what does `AlertsTableQueryContext` do?

Reply:
It was added in this [PR](https://github.com/elastic/kibana/pull/170651/files#diff-7e6e6c5bbe54f3c9e0dd82db4b1c69d1c49cafabe30bf68c69be7f344839ff6cR34). Perhaps @umbopepato or @Zacqary have more insight on why `AlertsTableQueryContext` is needed?

Reply:
We had to add it because in some use cases where the alerts table-related hooks were used inside nested `<QueryClientProviders>` we weren't able to pick up the right `QueryClient` instance to invalidate queries and refetch correctly: it was merely a matter of disambiguating queryClient instances.

To avoid duplications I guess we could export from the `triggersActionsUI` plugin an `<AlertsTableQueryClientProvider>` that automatically sets up a queryClient with the right context so that these hooks can be used unchanged outside of the AlertsTable as well 🙂